### PR TITLE
crosscluster/logical: implement new hook for new syntax

### PR DESCRIPTION
**Pull request by dt**

### CODE REVIEW

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go [hunk 1]
```diff
@@ -22,17 +37,158 @@ func init() {
...
+				tree.Name(td.GetName()),
+			)
+
+			srcTableNames[i] = stmt.On.Tables[i].String()
+
+			if i == 0 {
+				targetsDescription = tbNameWithSchema.FQString()
+			} else if i < 5 {
+				targetsDescription += ", " + tbNameWithSchema.FQString()
+			} else if i == 5 {
```
Comment (Jeremyyang920) [inline pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:139]: Is this just arbitrarily chosen?

**Review by Jeremyyang920 — APPROVED** [review]
LGTM!
I guess with the plan now in, the LDR script can also update since the test is now updated.

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go (hunk 1 above)
Comment (dt) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:139]: Yeah, we don’t like long job descriptions

Comment (dt) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go]: Actually just remembered that I changed `SHOW JOBS` to truncate long descriptions to 100 chars at display time so I removed the 5 cap here.

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go [hunk 2]
```diff
@@ -22,17 +37,156 @@ func init() {
...
+				TargetClusterConnStr: string(streamAddress),
+				ReplicationPairs:     repPairs,
+				TableNames:           srcTableNames},
+			Progress: jobspb.LogicalReplicationProgress{},
+		}
+
+		if _, err := p.ExecCfg().JobRegistry.CreateAdoptableJobWithTxn(ctx, jr, jr.JobID, p.InternalSQLTxn()); err != nil {
+			return err
+		}
+		p.ExecCfg().JobRegistry.NotifyToAdoptJobs()
```
Comment (stevendanna) [inline pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:170]: I know this was in the existing code, but I think this was an artifact of me not thinking things through. The job is created only when the txn commits so I don't think this does much for us.

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go [hunk 3]
```diff
@@ -22,17 +37,156 @@ func init() {
...
+		if err != nil {
+			return err
+		}
+		defer func() {
+			_ = client.Close(ctx)
+		}()
+
+		if err := client.Dial(ctx); err != nil {
+			return err
+		}
```
Comment (stevendanna) [inline pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:107]: Should we do all of the local work before connecting to the remote cluster?

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go [hunk 4]
```diff
@@ -22,17 +37,156 @@ func init() {
...
+
+		for i, name := range srcTableNames {
+			repPairs[i].SrcDescriptorID = int32(spec.TableDescriptors[name].ID)
+		}
+
+		jr := jobs.Record{
+			JobID:       p.ExecCfg().JobRegistry.MakeJobID(),
+			Description: fmt.Sprintf("LOGICAL REPLICATION STREAM into %s", targetsDescription),
```
Comment (stevendanna) [inline pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:155]: Would it be helpful to put the source cluster URL here?

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go [hunk 5]
```diff
@@ -396,10 +396,10 @@ func TestPreviouslyInterestingTables(t *testing.T) {
 				sqlA, tableName, numInserts, nil)
 			require.NoError(t, err)
 			streamStartStmt := fmt.Sprintf(
-				"SELECT crdb_internal.start_logical_replication_job('%s', ARRAY['%s'])",
-				dbAURL.String(), tableName)
+				"CREATE LOGICAL REPLICATION STREAM FROM $1 ON TABLE %s INTO TABLE %s",
+				tableName, tableName)
```
Comment (stevendanna) [inline pkg/ccl/crosscluster/logical/logical_replication_job_test.go:400]: I agree with the sentiment you've expressed elswhere that `ON` and `FROM` seem backwards to my reading. `FROM <table> ON <cluster>`.

**Review by stevendanna — COMMENTED** [review]
Thanks! Left a few minor questions for your consideration.

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go (hunk 4 above)
Comment (dt) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:155]: I'll put the Host, similar to PCR; people freak out if the keys are visible.

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go (hunk 3 above)
Comment (dt) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:107]: Done.

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go (hunk 5 above)
Comment (dt) [reply pkg/ccl/crosscluster/logical/logical_replication_job_test.go:400]: Let's get @alicia-l2's confirmation and then I'll open a follow-up PR swapping the order if she agrees too.

Comment (alicia-l2) [reply pkg/ccl/crosscluster/logical/logical_replication_job_test.go:400]: I'm fine with that

Comment (dt) [reply pkg/ccl/crosscluster/logical/logical_replication_job_test.go:400]: great, I'll do it in a follow-up.

**Comment by dt** [comment]
TFTR!

bors r+

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go (hunk 5 above)
Comment (dt) [reply pkg/ccl/crosscluster/logical/logical_replication_job_test.go:400]: follow-up is here: https://github.com/cockroachdb/cockroach/pull/126402

**Pull request merged**